main/selectors.c

```diff
@@ -81,11 +101,48 @@ tastePerlLine (const char *line)
 const char *
 selectByPickingPerlVersion (FILE *input)
 {
-    char line[0x800];
-    while (fgets(line, sizeof(line), input)) {
-        const char *lang = tastePerlLine(line);
-        if (lang)
-            return lang;
+    /* Default to Perl 5 */
+    return selectByLines (input, tastePerlLine, TR_PERL5);
+}
+
+static const char *
+tasteObjectiveCOrMatLabLines (const char *line)
+{
+#define TR_OBJC "ObjectiveC"
+#define TR_MATLAB "MatLab"
```

Comment:
Is using a `#define` better than `static const`?
Since you don't have a corresponding `#undef` at the end of the file, isn't there a (very low) risk that this define affects other files?